src/__tests__/to-be-visible.js

```diff
@@ -247,6 +247,185 @@ describe('.toBeVisible', () => {
           ).toBeVisible()
         })
       })
+
+      describe('when the details inner text does not have an enclosing element', () => {
+        describe('when the details is not opened', () => {
+          beforeEach(() => {
+            subject = render(`
+              <details>
+                <summary>Title of hidden innerText</summary>
+                hidden innerText
+              </details>
+            `)
+          })
+
+          it('returns false to the details content', () => {
+            expect(subject.container.querySelector('details')).not.toBeVisible()
+          })
+
+          it('returns true to the details summary', () => {
+            expect(subject.container.querySelector('summary')).toBeVisible()
+          })
+
+          describe('when the user clicks on the summary', () => {
+            beforeEach(() => subject.container.querySelector('summary').click())
+
+            it('returns true to the details content', () => {
+              expect(subject.container.querySelector('details')).toBeVisible()
+            })
+
+            it('returns true to the details summary', () => {
+              expect(subject.container.querySelector('summary')).toBeVisible()
+            })
+          })
+        })
+
+        describe('when the details is opened', () => {
+          beforeEach(() => {
+            subject = render(`
+              <details open>
+                <summary>Title of visible innerText</summary>
+                visible <small>innerText</small>
+              </details>
+            `)
+          })
+
+          it('returns true to the details content', () => {
+            expect(subject.container.querySelector('details')).toBeVisible()
+          })
+
+          it('returns true to inner small content', () => {
+            expect(subject.container.querySelector('small')).toBeVisible()
+          })
+
+          describe('when the user clicks on the summary', () => {
+            beforeEach(() => subject.container.querySelector('summary').click())
+
+            it('returns false to the details content', () => {
+              expect(
+                subject.container.querySelector('details'),
+              ).not.toBeVisible()
+            })
+
+            it('returns false to the inner small content', () => {
+              expect(subject.container.querySelector('small')).not.toBeVisible()
+            })
+
+            it('returns true to the details summary', () => {
+              expect(subject.container.querySelector('summary')).toBeVisible()
+            })
+          })
+        })
+
+        describe('with nested details (unenclosed outer, enclosed inner)', () => {
```

Comment:
This set of tests might be a bit redundant, I added them because I wasn't sure how the unenclosed and enclosed details would interact. But thinking it over, there really shouldn't be a significant difference as opposed to the previous nested tests (which these are based off).